CRM/Financial/BAO/FinancialTypeAccount.php

```diff
@@ -157,30 +151,20 @@ public static function del($financialTypeAccountId, $accountId = NULL) {
    * @param int $paymentInstrumentValue
    *   Payment instrument value.
    *
-   * @return array|null|string
+   * @return null|int
    */
-  public static function getInstrumentFinancialAccount($paymentInstrumentValue = NULL) {
-    if (!self::$financialAccount) {
-      $query = "SELECT ceft.financial_account_id, cov.value
-FROM civicrm_entity_financial_account ceft
-INNER JOIN civicrm_option_value cov ON cov.id = ceft.entity_id AND ceft.entity_table = 'civicrm_option_value'
-INNER JOIN civicrm_option_group cog ON cog.id = cov.option_group_id
-WHERE cog.name = 'payment_instrument' ";
-
-      if ($paymentInstrumentValue) {
-        $query .= " AND cov.value = '{$paymentInstrumentValue}' ";
-        return CRM_Core_DAO::singleValueQuery($query);
-      }
-      else {
-        $result = CRM_Core_DAO::executeQuery($query);
-        while ($result->fetch()) {
-          self::$financialAccount[$result->value] = $result->financial_account_id;
-        }
-        return self::$financialAccount;
-      }
-    }
-
-    return $paymentInstrumentValue ? self::$financialAccount[$paymentInstrumentValue] : self::$financialAccount;
+  public static function getInstrumentFinancialAccount($paymentInstrumentValue) {
+    $paymentInstrument = civicrm_api3('OptionValue', 'getsingle', array(
```

Comment:
@pradpnayak this is what is causing the test to fail would there be a reason why there is 2 payment_instruments with the same option value? Maybe its a test problem